orion-ui/src/compositions/useToast.ts

```diff
@@ -0,0 +1,19 @@
+import { showToast, Toast, ToastOptions, ToastType } from '@prefecthq/prefect-design'
+import { onUnmounted, Component } from 'vue'
+
+/**
+ * The useToast composition takes the same arguments as showToast but also dismisses the toast when the caller component is unmounted
+ * @param message string | Component
+ * @param type ToastType
+ * @param options ToastOptions
+ * @returns Toast
+ */
+export function useToast(message: string | Component, type?: ToastType, options?: ToastOptions): Toast {
```

Comment:
I think this composition is slightly wrong. 

since `useToast` has an `onUnmounted` hook it needs to be called from a setup function, not necessarily at the same time the developer wants a toast message to appear. 

I would have expected this composition to return a function, so it would always be "initiated" in the setup and called whenever the developer needs the toast to show up. 

<script lang="ts" setup>
    // no options needed 
    const showToast = useToast()
    
    async function doSomethingAsync(): Promise<void> {
        await something()
        // options go here like regular toast
        showToast(....)
    }
</script>

Reply:
+1

Reply:
Good call - let me see if I can find a better way to surface that

Reply:
Made an update to that, lmk what you think